rados/rados.go

```diff
@@ -80,7 +81,17 @@ func (c *RadosConn) newRadosConn() (*rados.Conn, error) {
 		return nil, fmt.Errorf("error setting rados_mon_op_timeout: %s", err)
 	}
 
-	err = conn.Connect()
+	ch := make(chan error, 1)
+	go func(conn *rados.Conn) {
+		ch <- conn.Connect()
```

Comment:
This should close `ch` as well, I think.

Reply:
Yep, but I had to do it from inside the goroutine